Common/src/Pollster/CreateLargeTaskProcessor.cs

```diff
@@ -0,0 +1,188 @@
+// (at your option) any later version.
+// 
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+// 
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+using ArmoniK.Api.gRPC.V1;
+using ArmoniK.Core.Common.gRPC.Services;
+using ArmoniK.Core.Common.StateMachines;
+using ArmoniK.Core.Common.Utils;
+
+using Microsoft.Extensions.Logging;
+
+using TaskOptions = ArmoniK.Api.gRPC.V1.TaskOptions;
+using TaskRequest = ArmoniK.Core.Common.gRPC.Services.TaskRequest;
+
+namespace ArmoniK.Core.Common.Pollster;
+
+internal class CreateLargeTaskProcessor : IProcessReplyProcessor
+{
+  private readonly ISubmitter                               submitter_;
+  private readonly IAsyncPipe<ProcessReply, ProcessRequest> asyncPipe_;
+  private readonly ILogger                                  logger_;
+  private          IList<string>?                           taskIds_;
+  private          TaskOptions?                             options_;
+  private          string?                                  requestId_;
+  private readonly string                                   sessionId_;
+  private readonly string                                   parentTaskId_;
+  private readonly ProcessReplyCreateLargeTaskStateMachine  fsm_;
+  private readonly Channel<TaskRequest>                     taskRequestsChannel_ = Channel.CreateBounded<TaskRequest>(10);
+  private          Channel<ReadOnlyMemory<byte>>?           payloadsChannel_;
+  private          Task?                                    completionTask_;
+
+  public CreateLargeTaskProcessor(ISubmitter                               submitter,
+                                  IAsyncPipe<ProcessReply, ProcessRequest> asyncPipe,
+                                  string                                   sessionId,
+                                  string                                   parentTaskId,
+                                  ILogger                                  logger)
+  {
+    submitter_       = submitter;
+    asyncPipe_       = asyncPipe;
+    logger_          = logger;
+    sessionId_       = sessionId;
+    parentTaskId_    = parentTaskId;
+    requestId_       = null;
+    fsm_             = new ProcessReplyCreateLargeTaskStateMachine(logger);
+  }
+
+  public async Task AddProcessReply(ProcessReply      processReply,
+                              CancellationToken cancellationToken)
+  {
+    if (requestId_ == null)
+    {
+      requestId_ = processReply.RequestId;
+    }
+    else
+    {
+      if (processReply.RequestId != requestId_)
```

Comment:
Is it possible for this two variables to have a different value? If yes, how?

Reply:
By construction, they are the same so its changed